CCVA_Processing_v2.Rmd

```diff
@@ -0,0 +1,2003 @@
+---
+title: "CCVA_Processing"
+author: "KEC"
+date: "2024-04-04"
+output: html_document
+editor_options: 
+  chunk_output_type: console
+---
+
+## CCVA Data Processing
+
+This R markdown prepares data inputs for a Climate Change Vulnerability 
+Assessment (CCVA) for water supplies at a selected National Park.
+
+### 1. Setup Workspace
+
+First, install and/or load required packages and functions.
+
+```{r setup, include=FALSE}
+
+knitr::opts_chunk$set(echo = TRUE)
+
+source('setup.R')
+library("readxl")
+library("trend")
+
+```
+
+
+### 2. Water Supply Database
+
+A water supply system database was created for this project using various NPS
+and public databases.  Start by importing the water supply table as a data 
+table that contains a row for each water supply system at the park. Then import
+the water source table which contains a row for each source of water (typically
+individual wells, diversions, and springs) that source each water supply. The
+two tables are related by (shared column) the CSU-assigned wsd_system_id.
+
+```{r, eval = TRUE, echo = FALSE, message = FALSE, warning = FALSE}
+
+
+table_path <- "data/Water_Supply_Systems/NPS_Water_Systems_Database.xlsx"
+
+# Supply table (Currently filtered to Utah)
+supply_table <- read_excel(table_path, sheet = 2, na = "NA") %>%
+  janitor::clean_names() %>%
+  dplyr::filter(region == "Intermountain Region",
+                state == "UT")
+  
+# Source table
+source_table <- read_excel(table_path, sheet = 3, na = "NA") %>%
+  janitor::clean_names() %>%
+  dplyr::filter(wsd_system_id %in% supply_table$wsd_system_id) %>%
+  dplyr::mutate(well_depth = as.numeric(well_depth))
+
+# Copy of source table as sf_object - some columns without location data are
+# dropped. Circle back to this later.
+source_table_locs <- source_table %>%
+  drop_na(c("source_longitude", "source_longitude")) %>%
+  st_as_sf(., 
+           coords = (c("source_longitude","source_latitude")), 
+           crs = 4326,
+           remove = FALSE)
+
+# Optional preview water source locations
+mapviewOptions(fgb = FALSE, 
+               georaster = FALSE, 
+               basemaps = c("Esri.WorldTopoMap", 
+                            "Esri.WorldImagery"))
+mapview(source_table_locs,
+        zcol = "type",
+        layer.name = "Source Type")
+
+```
+
+
+### 3. Source Water 
+
+```{r source_water}
+
+# Get source
+get_aoi_source <- function(df) {
```

Comment:
OOh it's the other one. TBH CCVA_Processing_v2.Rmd is just a placeholder for now because I switched fully to deep dive. I should probably just delete everything in there.